src/app/features/courses/components/courses-list/components/courses-search/courses-search.component.ts

```diff
@@ -1,21 +1,47 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  Component,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { fromEvent, map, filter, debounceTime, Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-courses-search',
   templateUrl: './courses-search.component.html',
   styleUrls: ['./courses-search.component.scss'],
 })
-export class CoursesSearchComponent implements OnInit {
-  searchValue: string = '';
+export class CoursesSearchComponent implements OnInit, OnDestroy {
+  private searchElement!: HTMLInputElement | null;
+  private subs: Subscription[] = [];
 
   @Output()
   searchValueChange = new EventEmitter<string>();
 
   constructor() {}
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.searchElement = document.querySelector('.search__input');
```

Comment:
This is an absolutely unacceptable way to call HTML el. Forget about JS DOM manipulations in Angular. This is strictly prohibited and bad code smell. 

Read about [@ViewChild](https://blog.angular-university.io/angular-viewchild/), @ViewChildren and [template variables](https://angular.io/guide/template-reference-variables).

You don't need ngOnInit hook for it as well.

Reply:
done